src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty.java

```diff
@@ -0,0 +1,284 @@
+package org.csanchez.jenkins.plugins.kubernetes;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import com.cloudbees.hudson.plugins.folder.AbstractFolder;
+import com.cloudbees.hudson.plugins.folder.AbstractFolderProperty;
+import com.cloudbees.hudson.plugins.folder.AbstractFolderPropertyDescriptor;
+
+import hudson.Extension;
+import hudson.model.ItemGroup;
+import hudson.model.Descriptor.FormException;
+import hudson.slaves.Cloud;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+/**
+ * Provides folder level Kubernetes configuration.
+ */
+public class KubernetesFolderProperty extends AbstractFolderProperty<AbstractFolder<?>> {
+
+    private static final Logger LOGGER = Logger.getLogger(KubernetesFolderProperty.class.getName());
+
+    private Set<String> permittedClouds = new HashSet<>();
+
+    /**
+     * Constructor.
+     */
+    @DataBoundConstructor
+    public KubernetesFolderProperty() {
+    }
+
+    private Set<String> getPermittedClouds() {
+        return permittedClouds;
+    }
+
+    private Set<String> getInheritedClouds() {
+        Set<String> inheritedPermissions = new HashSet<>();
+        collectAllowedClouds(inheritedPermissions, getOwner().getParent());
+
+        return inheritedPermissions;
+    }
+
+    /**
+     * Called from Jelly.
+     * 
+     * @return
+     */
+    public List<UsagePermission> getEffectivePermissions() {
+        List<UsagePermission> ps = buildPermissions();
+
+        Set<String> inheritedClouds = getInheritedClouds();
+
+        for (UsagePermission p : ps) {
+            if (inheritedClouds.contains(p.name)) {
+                p.setGranted(true);
+                p.setInherited(true);
+            }
+
+            if (getPermittedClouds().contains(p.name)) {
+                p.setGranted(true);
+            }
+        }
+
+        // a non-admin User is only allowed to see granted clouds
+        if (!userHasAdministerPermission()) {
+            ps = ps.stream().filter(f -> f.granted).collect(Collectors.toList());
+        }
+
+        return ps;
+    }
+
+    private static List<UsagePermission> buildPermissions() {
+        List<UsagePermission> ps = new ArrayList<>();
+
+        for (KubernetesCloud kubernetesCloud : getUsageRestrictedKubernetesClouds()) {
+            UsagePermission p = new UsagePermission();
+
+            String name = kubernetesCloud.name;
+            p.setName(name);
+            p.setGranted(false);
```

Comment:
not necessary, a `boolean` is false by default.

Reply:
fixed